### ISSUE TITLE
[Documentation] Linux Server capabilty

### ISSUE THREAD

**Issue by spannerman79** [Erdroy/Stationeers.Addons#10]
I read the `README.md` and there was no reference in regards to how this would affect/work with a dedicated server.

Now I am going to assume that you have **both** linux and windows dedicated servers due to the fact that Stationseers dedicated server can be ran on both.

The reason why I opened this issue was specifically because of https://github.com/Erdroy/Stationeers.Addons/blob/master/README.md?plain=1#L9

For a linux dedicated server there is no GUI - hence no "right click X, blah blah blah"

NB: Yes I know how to run a dedicated server, Stationeer amongst others.  Your docs only reflect a **client** not a server.  If this is to be the defacto system to use for modded Stationeers this needs to be addressed.

**Comment by Erdroy** [comment]
Hi!

Yes, we're missing dedicated server docs because we do not support Linux currently.
Windows dedicated servers should work fine, and as far as I can tell, some people already been using it.

The issue with Linux support is that we need to run plugin compiler under mono runtime. 
Here is the whole description:
Copy-paste:
`The only issue why this will not work under linux hood, is because we're hosting the plugin compiler as so:
https://github.com/Erdroy/Stationeers.Addons/blob/master/Source/Stationeers.Addons/Modules/Plugins/PluginCompiler.cs#L23 
This tries to run the exe file just like on windows, and linux fails to do that - it would have to run it through mono runtime.
Maybe this could be as simple to solve as to just detect if running under Linux and then use mono executable for that and pass the path etc. as arguments (you would have to install mono-runtime).` [Discord message](https://ptb.discord.com/channels/795601381956124693/795601381956124696/882540660060737577)

To install this on Linux machine (for the game or server), you have to run patcher file manually using mono. But plugin compiler won't run, so it will error-our during plugin compilation.

Once we get higher priority tasks down (AST validation/whitelisting and resolve the code injection issue with custom content [you can inject some callbacks into some types of assets - that can lead to RCE]) - then this will be the time for making Addons a bit more usable and dedicated server friendly. 

I'll leave this issue open for now.

**Comment by spannerman79** [comment]
> To install this on Linux machine (for the game or server)

The only issue I see for it for a linux client is that (currently) I can easily play Stationeers via Proton.  So in order for the Linux playerbase (not much, but did reach 1%, but 1% of Steam Playerbase = ~ 1,204,000 users - [source](https://www.gamingonlinux.com/2021/09/linux-continues-to-remain-above-1-on-the-steam-hardware-survey)) it would be _nice_ if this doesn't have any issues being used via Proton.

**Comment by Gfurst** [comment]
Just came to ask and point out, that apparently the loader doesn't work via Proton, as stated above. But from what I understood on the OP, its just a suggestion to add proper documentation. 
Do I need to get a stand alone server running or is it possible to get the loader running through wine? aka, it could just something missing in the loading chain.

**Comment by specs32** [comment]
> > To install this on Linux machine (for the game or server)
> 
> The only issue I see for it for a linux client is that (currently) I can easily play [Stationeers via Proton](https://www.protondb.com/app/544550). So in order for the Linux playerbase (not much, but did reach 1%, but 1% of Steam Playerbase = ~ 1,204,000 users - [source](https://www.gamingonlinux.com/2021/09/linux-continues-to-remain-above-1-on-the-steam-hardware-survey)) it would be _nice_ if this doesn't have any issues being used via Proton.

I totally agree, it would be nice, really really nice ! :)  And as a linux user I would totally be okay with the fact that you need to run a script before playing to activate and integrate the modifications to the game or something similar.. To be able to just install the Stationeers.Addons via workshop and simply use them with proton would be the golden cradle ;) Where is the tip jar ?
